src/wagtailtrans/views/translation.py

```diff
@@ -1,63 +1,45 @@
 from __future__ import absolute_import, unicode_literals
 
+from django.core.urlresolvers import reverse
 from django.shortcuts import get_object_or_404, redirect
 from django.utils.translation import ugettext_lazy as _
-from django.views.generic.edit import FormView
-from wagtail.wagtailadmin.edit_handlers import (
-    FieldPanel, ObjectList, PageChooserPanel, TabbedInterface)
+from wagtail.wagtailadmin.views.generic import CreateView
 
 from wagtailtrans.forms import TranslationForm
 from wagtailtrans.models import Language, TranslatablePage
 
 
-class Add(FormView):
-    """View to add a new ``Translation``."""
-
+class TranslationView(CreateView):
+    model = TranslatablePage
     form_class = TranslationForm
-    template_name = 'wagtailtrans/translation/add.html'
-
-    edit_handler = TabbedInterface([
-        ObjectList(
-            [
-                FieldPanel('copy_from_canonical'),
-                PageChooserPanel('parent_page'),
-            ],
-            heading=_("Translate"),
-            base_form_class=TranslationForm
-        ),
-    ])
-
-    def dispatch(self, request, page_pk, language_code, *args, **kwargs):
-        self.page = get_object_or_404(TranslatablePage, pk=page_pk).specific
+
+    def page_title(self):
+        return _("Create {} in {}".format(self.instance.title, self.language))
```

Comment:
Shouldn't this be something like ``get_admin_title``? With the prospect that it can also be a [draft title](https://github.com/wagtail/wagtail/pull/3285) if this gets merged?